public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -288,6 +289,25 @@ export class LokiDatasource
     return { ...logsSampleRequest, targets };
   }
 
+  private getQueryHeaders(request: DataQueryRequest<LokiQuery>): Record<string, string> {
+    const headers: Record<string, string> = {};
+    if (!config.featureToggles.lokiSendDashboardPanelNames) {
+      return headers;
+    }
+
+    const dashboard = getDashboardSrv().getCurrent();
```

Comment:
check what happens in explore -- you may want to break early here, or return something else (maybe look at the URL?)